# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessibleBase.cpp
# Commit: afea577be988
# Full Hash: afea577be988aff29d895b25c54e20a4b4420513
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 2
# Description:
#   Bug 1737944 - P3: Introduce Accessible::TagName getter for retrieving HTML tags. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134206
# ==============================================================================

diff -r 7e586e1f7892 -r afea577be988 accessible/ipc/RemoteAccessibleBase.cpp
--- a/accessible/ipc/RemoteAccessibleBase.cpp	Tue Jan 04 21:01:34 2022 +0000
+++ b/accessible/ipc/RemoteAccessibleBase.cpp	Tue Jan 04 21:01:35 2022 +0000
@@ -489,6 +489,18 @@
 }
 
 template <class Derived>
+nsAtom* RemoteAccessibleBase<Derived>::TagName() const {
+  if (mCachedFields) {
+    if (auto tag =
+            mCachedFields->GetAttribute<RefPtr<nsAtom>>(nsGkAtoms::tag)) {
+      return *tag;
+    }
+  }
+
+  return nullptr;
+}
+
+template <class Derived>
 void RemoteAccessibleBase<Derived>::TakeFocus() const {
   Unused << mDoc->SendTakeFocus(mID);
 }